src/librustc/mir/repr.rs

```diff
@@ -62,7 +63,7 @@ macro_rules! newtype_index {
 pub struct Mir<'tcx> {
     /// List of basic blocks. References to basic block use a newtyped index type `BasicBlock`
     /// that indexes into this vector.
-    basic_blocks: IndexVec<BasicBlock, BasicBlockData<'tcx>>,
+    pub basic_blocks: IndexVec<BasicBlock, BasicBlockData<'tcx>>,
```

Comment:
Still can't do that, for the same reason I can't use the accessor methods. I'm moving stuff out of the struct, so I wouldn't be able to move the struct into a `take` method. Unless the method unpacked the *entire* MIR.